lua/entities/gmod_wire_textentry.lua

```diff
@@ -164,23 +171,36 @@ net.Receive("wire_textentry_action",function(len,ply)
 	self:Unprompt() -- in all cases, make text entry available for use again
 
 	if ok and not self.BlockInput then
-		WireLib.TriggerOutput( self, "Text", text )
+		self:OnTextEntered(text)
 
-		local timername = "wire_textentry_" .. self:EntIndex()
-		timer.Remove( timername )
-		if math.max(self:GetHold(),0) > 0 then
-			timer.Create( timername, math.max(self:GetHold(),0), 1, function()
-				if IsValid( self ) then
-					WireLib.TriggerOutput( self, "User", nil )
-					WireLib.TriggerOutput( self, "Text", "" )
-				end
-			end)
-		end
+		
 	end
 
 	self:UpdateOverlay()
 end)
 
+local ENTERED_DISABLE_DELAY = 0.1
+
+function ENT:OnTextEntered(text)
+	WireLib.TriggerOutput( self, "Text", text )
+	WireLib.TriggerOutput( self, "Entered", 1 )
+
+	local timername = "wire_textentry_" .. self:EntIndex()
+	timer.Remove( timername )
+	if math.max(self:GetHold(),0) > 0 then
+		timer.Create( timername, math.max(self:GetHold(),0), 1, function()
+			if IsValid( self ) then
+				WireLib.TriggerOutput( self, "User", nil )
+				WireLib.TriggerOutput( self, "Text", "" )
+			end
+		end)
+	end
+
+	timer.Create(timername.."_disable_entered", ENTERED_DISABLE_DELAY, 1, function()
+		WireLib.TriggerOutput( self, "Entered", 0 )
+	end)
```

Comment:
Is a hold really necessary?
You could just do this. Also text output doesn't need to be reset.
```lua
WireLib.TriggerOutput( self, "Text", text )
WireLib.TriggerOutput( self, "Entered", 1 )
WireLib.TriggerOutput( self, "Entered", 0 )
```

Reply:
It is from original code. I just refactored it into separate function and added `_disable_entered` timer.